src/ThreeBRSSyliusPaymentRestrictionPlugin.php

```diff
@@ -7,7 +7,12 @@
 use Sylius\Bundle\CoreBundle\Application\SyliusPluginTrait;
 use Symfony\Component\HttpKernel\Bundle\Bundle;
 
-class ThreeBRSSyliusPaymentRestrictionPlugin extends Bundle
+final class ThreeBRSSyliusPaymentRestrictionPlugin extends Bundle
 {
     use SyliusPluginTrait;
+
+    public function getPath(): string
+    {
+        return \dirname(__DIR__);
+    }
```

Comment:
proč je tu tohle?

Reply:
Tak to mi teď zabralo docela dlouho, než jsem to rozklíčoval:D

- pochází to z oficiálního Sylius plugin skeletonu https://github.com/Sylius/PluginSkeleton/blob/1.12/src/AcmeSyliusExamplePlugin.php
- požaduje to tak Symfony https://symfony.com/doc/current/bundles/best_practices.html#directory-structure _This directory structure requires to configure the bundle path to its root directory as follows:_
```php
class AcmeBlogBundle extends Bundle
{
    public function getPath(): string
    {
        return \dirname(__DIR__);
    }
}
```
- načítá to Kernel při obnově cache v `\Symfony\Component\HttpKernel\Kernel::getKernelParameters`
- řekl bych, že je v tom krapet bordel, protože podle toho co mi prozradil XDebug, tak některé bundles ukazují do svého root adresáře, ale jiné do `src` adresáře s PHP classes

Navrhuji nechat jak je teď - za prvé to funguje a za druhé to tak chce Symfony documentace.

PS: moc nevím, k čemu to Symfony používá. Možná k překladu logic path na absolute path https://stackoverflow.com/questions/14679377/get-the-path-of-a-directory-inside-a-bundle-in-symfony

Reply:
Tak ne. `3brs-sylius-sorting-plugin` ukázal, že správně je původní `\Symfony\Component\HttpKernel\Bundle\Bundle::getPath`, což ukazuje někam do `src`. Když jsem tam dal `dirname(__DIR__)`, tak Symfony nedokázala najít logic path `@ThreeBRSSyliusSortingPlugin/Resources/config/routing.yml`

Možná že se starou Symfony strukturou "všechno v src" je správně původní `Bundle\Bundle::getPath` a teprve s novou flat strukturou tak jak to má Sylius skeleton plugin :thinking: 

Každopádně jsem tohle smazal.